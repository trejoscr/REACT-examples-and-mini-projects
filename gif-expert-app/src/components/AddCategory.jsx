import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

  const [inpuValue, setInpuValue] = useState('');

  const onInputChange = ({target}) => {
    setInpuValue(target.value);
  }

  const onSubmit = (event) => {

    event.preventDefault();    
    if(inpuValue.trim().length <= 1) return

    //setCategories(categories => [inpuValue, ...categories]);
    
    setInpuValue('');
    onNewCategory(inpuValue.trim());
    
  }

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
      type="text"
        placeholder="Search gifs..." 
        value={inpuValue}
        onChange={onInputChange}
      />
    </form>    
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}