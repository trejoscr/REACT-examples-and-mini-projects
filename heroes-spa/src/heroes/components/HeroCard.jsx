import { Link } from "react-router-dom"

export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters
}) => {

  const heroImageUrl = `/heroes/${id}.jpg`;

  const charactersByHero = (<p>{characters}</p>);

  return (
    <div className="col animate__animated animate__fadeIn">
      <div className="card">

        <div className="row no-gutters">

          <div className="col-12 col-lg-4">
            <img src={heroImageUrl} alt={superhero} className="card-img"/>
          </div>

          <div className="row-12 col-lg-8 p-4 p-lg-2">
            <h5>{superhero}</h5>
            <p className="card-text">{alter_ego}</p>
            {
              (alter_ego !== characters) && charactersByHero
            }

            <p className="card-text">
              <small>{first_appearance}</small>
            </p>

            <Link to={`/hero/${id}`}>
              More ...
            </Link>
            
          </div>
          
        </div>

      </div>
    </div>
  )
}
