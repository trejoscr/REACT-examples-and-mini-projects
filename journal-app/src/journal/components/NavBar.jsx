import { LogoutOutlined, MenuOpenOutlined } from "@mui/icons-material"
import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material"
import { useDispatch } from "react-redux"
import { startLogout } from "../../store/auth/thunks";

export const NavBar = ({drawerWidth = 240}) => {

  const dispatch = useDispatch();

  const onLogout = () => {
    dispatch(startLogout());
  }

  return (
    <AppBar
      position='fixed'
      sx={{
        width:{sm: `calc(100% - ${drawerWidth}px)`},
        ml:{sm:`${drawerWidth}px`}
      }}
    >
      <Toolbar>
        <IconButton
          color='inherit'
          edge="start"
          sx={{mr:2, display:{sm: 'none'}}}
        >
          <MenuOpenOutlined/>
        </IconButton>

        <Grid container direction='row' justifyContent='space-between' alignItems='center'>
          <Typography variant="h6" noWrap component='div'>JournalAPP</Typography>
          <IconButton color='inherit' onClick={onLogout}>
            <LogoutOutlined/>
          </IconButton>
        </Grid>

      </Toolbar>
    </AppBar>
  )
}
