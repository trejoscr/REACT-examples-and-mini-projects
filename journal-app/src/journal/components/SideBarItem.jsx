import { TurnedInNot } from "@mui/icons-material"
import { Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
import { useMemo } from "react"
import { setActiveNote } from "../../store/journal/journalSlice";
import { useDispatch } from "react-redux";

export const SideBarItem = ({title = '', body, id, date, imageUrls = []}) => {

  const dispath = useDispatch();

  const onClickNote = () => {
    dispath(setActiveNote({title, body, id, date, imageUrls}));
  };  

  const newTitle = useMemo(()=>{
    return title?.length > 15 ? `${title.substring(0,15)}...` : title
  },[title])

  return (
    <ListItem disablePadding>
      <ListItemButton onClick={onClickNote}>
        <ListItemIcon>
          <TurnedInNot/>
        </ListItemIcon>
        <Grid container>
          <ListItemText primary={newTitle}/>
          <ListItemText secondary={body}/>
        </Grid>
      </ListItemButton>
    </ListItem>
  )
}