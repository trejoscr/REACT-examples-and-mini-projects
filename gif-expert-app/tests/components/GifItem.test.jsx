import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas es <GifItem/>', () => { 

  const title = 'Super Mario';
  const url = 'https://www.lavanguardia.com/andro4all/hero/2023/05/super-mario-bros-la-pelicula.png';

  test('hacer match con el snapshot', () => { 
    const {container} = render(<GifItem title={title} url={url}/>);
    expect(container).toMatchSnapshot();
  })

  test('debe de mostrar la imagen con el url y el alt indicado', () => { 
    
    render(<GifItem title={title} url={url}/>);
    //screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);

    const {src, alt} = screen.getByRole('img');
    expect(src).toEqual(url);
    expect(alt).toEqual(title);

  })

  test('debe de mostrar el titulo en el componente', () => { 
    render(<GifItem title={title} url={url}/>);
    expect(screen.getByText(title)).toBeTruthy();
  })

})