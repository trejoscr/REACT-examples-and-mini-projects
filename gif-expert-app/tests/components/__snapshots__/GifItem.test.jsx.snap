// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es <GifItem/> hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Super Mario"
      src="https://www.lavanguardia.com/andro4all/hero/2023/05/super-mario-bros-la-pelicula.png"
    />
    <p>
      Super Mario
    </p>
  </div>
</div>
`;
